sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/SamplingResult.java

```diff
@@ -23,48 +23,112 @@
+   *
+   * @param attributes The attributes to return from {@link SamplingResult#getAttributes()}. A
+   *     different object instance with the same elements may be returned.
+   * @return A {@link SamplingResult} with the attributes equivalent to {@code attributes} and the
+   * {@link SamplingDecision#RECORD_AND_SAMPLE}
+   */
+  static SamplingResult recordAndSample(Attributes attributes) {
+    requireNonNull(attributes, "attributes");
+    return attributes.isEmpty()
+        ? ImmutableSamplingResult.getEmptyRecordedAndSampledSamplingResult()
+        : ImmutableSamplingResult.createSamplingResult(
+            SamplingDecision.RECORD_AND_SAMPLE, attributes);
+  }
+
+  /**
+   * Returns a {@link SamplingResult} corresponding to {@link SamplingDecision#RECORD_ONLY} with
+   * no attributes and {@link SamplingResult#getDecision()} returning {@code decision}.
+   *
+   * <p>Use {@link #recordOnly(Attributes)} if you need attributes.
+   *
+   * <p>This is meant for use by custom {@link Sampler} implementations.
+   *
+   * @return A {@link SamplingResult} with empty attributes and the provided {@code decision}.
+   */
+  static SamplingResult recordOnly() {
+    return ImmutableSamplingResult.getEmptyRecordedSamplingResult();
+  }
+
+  /**
+   * Returns a {@link SamplingResult} corresponding to {@link SamplingDecision#RECORD_ONLY} with
+   * the given {@code attributes} and {@link SamplingResult#getDecision()} returning
+   * {@code decision}.
    *
    * <p>This is meant for use by custom {@link Sampler} implementations.
    *
-   * <p>Use {@link #create(SamplingDecision, Attributes)} if you need attributes.
+   * <p>Using {@link #recordOnly()}  instead of this method is slightly faster and
+   * shorter if you don't need attributes.
+   *
+   * @param attributes The attributes to return from {@link SamplingResult#getAttributes()}. A
+   *     different object instance with the same elements may be returned.
+   * @return A {@link SamplingResult} with the attributes equivalent to {@code attributes} and the
+   * {@link SamplingDecision#RECORD_ONLY}
+   */
+  static SamplingResult recordOnly(Attributes attributes) {
+    requireNonNull(attributes, "attributes");
+    return attributes.isEmpty()
+        ? ImmutableSamplingResult.getEmptyRecordedSamplingResult()
+        : ImmutableSamplingResult.createSamplingResult(SamplingDecision.RECORD_ONLY, attributes);
+  }
+
+  /**
+   * Returns a {@link SamplingResult} corresponding to {@link SamplingDecision#DROP} with
+   * no attributes and {@link SamplingResult#getDecision()} returning {@code decision}.
+   *
+   * <p>Use {@link #drop(Attributes)} if you need attributes.
+   *
+   * <p>This is meant for use by custom {@link Sampler} implementations.
    *
-   * @param decision The decision made on the span.
    * @return A {@link SamplingResult} with empty attributes and the provided {@code decision}.
    */
-  static SamplingResult create(SamplingDecision decision) {
```

Comment:
we can't delete this public method. That would be a breaking change.

Reply:
is it too early to deprecate?

Reply:
I think it's fine to keep it here. we could deprecate at some point, but it's not harmful or bad to use this method, just slightly confusing since it won't actually *create* anything.